BootloaderCommonPkg/Library/ContainerLib/ContainerLib.c

```diff
@@ -337,6 +429,69 @@ LocateComponentEntry (
   return EFI_SUCCESS;
 }
 
+/**
+  Get Next Component in a container.
+
+  @param[in]       ContainerSig    Container signature or component type.
+  @param[in, out]  ComponentName   Pointer to receive next component name.
+
+  @retval EFI_INVALID_PARAMETER    ComponentName is NULL.
+  @retval EFI_NOT_FOUND            Cannot find next component.
+  @retval EFI_SUCCESS              Next component was found successfully.
+
+**/
+EFI_STATUS
+EFIAPI
+GetNextComponent (
+  IN     UINT32     ContainerSig,
+  IN     UINT32    *ComponentName
+)
+{
+  EFI_STATUS         Status;
+  CONTAINER_HDR     *ContainerHdr;
+  CONTAINER_ENTRY   *ContainerEntry;
+  COMPONENT_ENTRY   *CurrEntry;
+  COMPONENT_ENTRY   *NextEntry;
+  UINT32             Index;
+
+  if (ComponentName == NULL) {
+    return EFI_INVALID_PARAMETER;
+  }
+
+  // Search container header from cache
+  Status = EFI_NOT_FOUND;
+  ContainerEntry = GetContainerBySignature (ContainerSig);
+  if (ContainerEntry == NULL) {
+    return Status;
+  }
+
+  ContainerHdr = (CONTAINER_HDR *)ContainerEntry->HeaderCache;
+  if (ContainerHdr->Count == 0) {
+    return Status;
+  }
+
+  CurrEntry = (COMPONENT_ENTRY *)&ContainerHdr[1];
+  for (Index = 0; Index < (UINT32)(ContainerHdr->Count - 1); Index++) {
+    NextEntry = (COMPONENT_ENTRY *)((UINT8 *)(CurrEntry + 1) + CurrEntry->HashSize);
+    if ((CurrEntry->Attribute & COMPONENT_ENTRY_ATTR_RESERVED) == 0) {
+      if ((NextEntry->Attribute & COMPONENT_ENTRY_ATTR_RESERVED) == 0) {
```

Comment:
Why current and next need to be checked?

Reply:
Good catch. Need to redo the code here. Basic intent is to support if there is only 1 component and second component is _SG_ (last component)